lib/cli/do_update_other.js

```diff
@@ -440,6 +464,28 @@ function do_update_other(subcmd, opts, args, cb) {
             }, next);
         },
 
+        function ensureServicesWantDelegatedDataset(ctx, next) {
+            vasync.forEachParallel({
+                inputs: ctx.svcs,
+                func: function updateSvc(svc, nextSvc) {
+                    if (WANT_DELEGATED_DATASET.indexOf(svc.name) !== -1 &&
+                        svc.type === 'vm' && svc.params &&
+                        svc.params.delegate_dataset !== true) {
+                        progress(
+                            'Add delegated dataset to  "%s" service ' +
+                            'definition. Dataset will be created on the next ' +
+                            'reprovision.',
+                            svc.name);
+                        updateService(svc.uuid,
+                            { params: { delegate_dataset: true } },
+                            nextSvc);
```

Comment:
Do we need to propagate errors here?  I see sdcClientErrWrap() in the stage above this one, but maybe this one doesn't error out?

Reply:
That's a good question...I basically copied this directly from the `updateMaintainResolvers` function starting on line 241. I think it would be much more work to go add that in to all the places that need it.

But, sdcadm has good logging, and uncaught errors will produce a dump with log information. So I think that's why there's not a lot of error handling in sdcadm. At any rate, because I'm following an established pattern I'm not *as* concerned about it for this particular commit.

Reply:
Works for me, thank you for the context.